connect_ext_ppr/tasks_manager.py

```diff
@@ -55,29 +51,34 @@ async def main_process(deployment_request_id, config):
             deployment_request=deployment_request.id,
         ).order_by(Task.id).all()
 
-        last_task_result = await execute_task(tasks[0], db, validate_ppr)
-        if last_task_result == TasksStatusChoices.ERROR:
-            deployment_request_final_status = DeploymentRequestStatusChoices.ERROR
-        else:
-            db.refresh(tasks[1])
-            last_task_result = await execute_task(
-                tasks[1],
-                db,
-                apply_ppr_and_delegate_to_marketplaces,
-            )
-            if last_task_result == TasksStatusChoices.ERROR:
-                deployment_request_final_status = DeploymentRequestStatusChoices.ERROR
-            elif deployment_request.delegate_l2:
-                db.refresh(tasks[2])
-                last_task_result = await execute_task(tasks[2], db, delegate_to_l2)
-                if last_task_result == TasksStatusChoices.DONE:
-                    deployment_final_status = DeploymentStatusChoices.SYNCED
-                else:
-                    deployment_request_final_status = DeploymentRequestStatusChoices.ERROR
-
-        deployment.status = deployment_final_status
-        deployment_request.status = deployment_request_final_status
+        deployment_request.status = DeploymentRequestStatusChoices.DONE
+        for task in tasks:
```

Comment:
I think you should have 2 separate transactions here. First - before you run the task --> commit; next run the task --> commit; and 3d one calculate update statuses --> commit; Otherwise the transaction is going to be opened for quite long time.